src/main/java/com/powsybl/openloadflow/network/PiModelArray.java

```diff
@@ -26,10 +26,13 @@ public class PiModelArray implements PiModel {
 
     private LfBranch branch;
 
+    private Direction r1Variation;
```

Comment:
Ideally we should not have anything release to past state inside pi model array but just a counter in the outer loop context.